### PR TITLE
Add more unit tests to AppView components

### LINKED ISSUE #306: [unit tests] AppView component
Add unit tests for AppView and related sub-components.

### PR DESCRIPTION
**Pull request by andresmgot**
Continuation of #527

Closes #306

Add basic renderization tests for the components of AppView that didn't have them yet.

### CODE REVIEW

**Comment by andresmgot** [comment]
The linter is complaining about wrong order. Hopefully I can rebase #544 here.

File: dashboard/src/components/AppView/AppView.test.tsx [hunk 1]
```diff
@@ -114,4 +122,36 @@ describe("AppViewComponent", () => {
       expect(sockets.length).toEqual(0);
     });
   });
+
+  describe("renderization", () => {
```
Comment (prydonius) [inline dashboard/src/components/AppView/AppView.test.tsx]: Any reason for nesting these?

File: dashboard/src/components/AppView/AppView.test.tsx [hunk 2]
```diff
@@ -114,4 +122,36 @@ describe("AppViewComponent", () => {
...
+    it("renders all the elements of an application", () => {
+      const wrapper = shallow(<AppViewComponent {...validProps} />);
+      expect(wrapper.find(ChartInfo).exists()).toBe(true);
+      expect(wrapper.find(DeploymentStatus).exists()).toBe(true);
+      expect(wrapper.find(AppControls).exists()).toBe(true);
+      expect(wrapper.find(ServiceTable).exists()).toBe(true);
+      expect(wrapper.find(AppNotes).exists()).toBe(true);
+      expect(wrapper.find(AppDetails).exists()).toBe(true);
+    });
```
Comment (prydonius) [inline dashboard/src/components/AppView/AppView.test.tsx]: can we separate these with new lines?

File: dashboard/src/components/AppView/AppView.test.tsx [hunk 3]
```diff
@@ -114,4 +122,36 @@ describe("AppViewComponent", () => {
...
+    it("renders a forbidden error if it exists", () => {
+      const wrapper = shallow(
+        <AppViewComponent {...validProps} deleteError={new ForbiddenError()} />,
+      );
+      expect(wrapper.find(PermissionsErrorAlert).exists()).toBe(true);
```
Comment (prydonius) [inline dashboard/src/components/AppView/AppView.test.tsx]: we can put the result of `wrapper.find` in a variable so we don't try to find it twice

File: dashboard/src/components/AppView/AppView.test.tsx [hunk 4]
```diff
@@ -114,4 +122,36 @@ describe("AppViewComponent", () => {
...
+    it("renders all the elements of an application", () => {
+      const wrapper = shallow(<AppViewComponent {...validProps} />);
+      expect(wrapper.find(ChartInfo).exists()).toBe(true);
+      expect(wrapper.find(DeploymentStatus).exists()).toBe(true);
+      expect(wrapper.find(AppControls).exists()).toBe(true);
+      expect(wrapper.find(ServiceTable).exists()).toBe(true);
+      expect(wrapper.find(AppNotes).exists()).toBe(true);
+      expect(wrapper.find(AppDetails).exists()).toBe(true);
+    });
+    it("renders a forbidden error if it exists", () => {
```
Comment (prydonius) [inline dashboard/src/components/AppView/AppView.test.tsx]: These last two tests are testing the displaying of the deleteError, what about if there is an error when initially fetching the application?

File: dashboard/src/components/AppView/DeploymentTable.test.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import * as React from "react";
+
+import { IResource } from "shared/types";
+import DeploymentTable from "./DeploymentTable";
+import DeploymentItem from "./DeploymentItem";
+
+it("renders a deployment ready", () => {
+  const deployments = new Map<string, IResource>();
+  const dep = "foo";
+  deployments[dep] = {
```
Comment (prydonius) [inline dashboard/src/components/AppView/DeploymentTable.test.tsx]: can we test with 2 deployments and ensure there are two DeploymentItem components rendered?

File: dashboard/src/components/AppView/ServiceTable.test.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  const service = {
+    metadata: {
+      name: "foo",
+    },
+    spec: {
+      type: "LoadBalancer",
+    },
+  } as IResource;
+  const services = new Map<string, IResource>();
+  services[service.metadata.name] = service;
```
Comment (prydonius) [inline dashboard/src/components/AppView/ServiceTable.test.tsx]: likewise with the DeploymentTable it might be cool to test multiple services

File: dashboard/src/components/AppView/AppView.test.tsx [hunk 7]
```diff
@@ -114,4 +122,50 @@ describe("AppViewComponent", () => {
...
+      const wrapper = shallow(<AppViewComponent {...validProps} />);
+      expect(wrapper.find(ChartInfo).exists()).toBe(true);
+      expect(wrapper.find(DeploymentStatus).exists()).toBe(true);
+      expect(wrapper.find(AppControls).exists()).toBe(true);
+      expect(wrapper.find(ServiceTable).exists()).toBe(true);
+      expect(wrapper.find(AppNotes).exists()).toBe(true);
+      expect(wrapper.find(AppDetails).exists()).toBe(true);
+    });
+
+    it("renders an error if it exists", () => {
```
Comment (prydonius) [inline dashboard/src/components/AppView/AppView.test.tsx]: I read this as "if the app exists" which obviously in this case the app doesn't exist, but I think you mean that the `error` exists. Maybe we can reword this to say "renders an error if error prop is set" to make it very clear?

File: dashboard/src/components/AppView/AppView.test.tsx [hunk 8]
```diff
@@ -114,4 +122,50 @@ describe("AppViewComponent", () => {
...
+      const wrapper = shallow(<AppViewComponent {...validProps} error={new NotFoundError()} />);
+      const err = wrapper.find(NotFoundErrorAlert);
+      expect(err.exists()).toBe(true);
+      expect(err.props()).toMatchObject({
+        namespace: "my-happy-place",
+        resource: 'Application "mr-sunshine"',
+      });
+    });
+
+    it("renders a forbidden delete-error if it exists", () => {
```
Comment (prydonius) [inline dashboard/src/components/AppView/AppView.test.tsx]: likewise here we can say "if the deleteError prop is a ForbiddenError"

File: dashboard/src/components/AppView/AppView.test.tsx [hunk 9]
```diff
@@ -114,4 +122,50 @@ describe("AppViewComponent", () => {
...
+      );
+      const err = wrapper.find(PermissionsErrorAlert);
+      expect(err.exists()).toBe(true);
+      expect(err.props()).toMatchObject({
+        action: 'delete Application "mr-sunshine"',
+        namespace: "my-happy-place",
+      });
+    });
+
+    it("renders a not-found delete-error if it exists", () => {
```
Comment (prydonius) [inline dashboard/src/components/AppView/AppView.test.tsx]: "if the deleteError prop is a NotFound error"

File: dashboard/src/components/AppView/DeploymentTable.test.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
+import { shallow } from "enzyme";
+import * as React from "react";
+
+import { IResource } from "shared/types";
+import DeploymentItem from "./DeploymentItem";
+import DeploymentTable from "./DeploymentTable";
+
+it("renders a deployment ready", () => {
```
Comment (prydonius) [inline dashboard/src/components/AppView/DeploymentTable.test.tsx]: I think we can combine this with the one about two deployments, but up to you!

Comment (andresmgot) [reply dashboard/src/components/AppView/DeploymentTable.test.tsx]: I want to keep this test as simple as possible because it's comparing the result with a snapshot. The other test is less strict about the layout as far as it renders two elements.
